### PR TITLE
[Docs] Add documentation for search queries in high-level rest client

### PR DESCRIPTION
**Pull request by tlrx**
This PR adds some documentation about how to build search queries with the Java High Level Rest Client. The idea is to have a simple mention on the Search API page, and then a complete table that links the search query as it is documented in the reference's Query DSL with the corresponding QueryBuilder class and QueryBuilders method, something like:

capture du 2017-07-31 16-50-18

### CODE REVIEW

File: docs/java-rest/high-level/apis/queries.asciidoc [hunk 1]
```diff
@@ -0,0 +1,24 @@
+[[java-rest-high-search-queries]]
+=== Search Queries
+
+This page lists all the available search type with their corresponding `QueryBuilder` class name and helper method name in the
```
Comment (javanna) [inline docs/java-rest/high-level/apis/queries.asciidoc]: I wouldn't want to mention "search type" in here. Let's call it queries?

File: docs/java-rest/high-level/apis/search.asciidoc [hunk 2]
```diff
@@ -82,8 +82,54 @@ After this, the `SearchSourceBuilder` only needs to be added to the
...
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-query-builder-options]
+--------------------------------------------------
+<1> Enable fuzzy matching on the match query
+<2> Set the prefix length option on the match query
+<3> Set the max expansion options to control the fuzzy
+    process of the query
+
+`QueryBuilder` objects can also be created using the `QueryBuilders` utility class.
+This class provides helper methods that can be used to create `QueryBuilder` objects
+ using a fluent programming style:
```
Comment (javanna) [inline docs/java-rest/high-level/apis/search.asciidoc]: I am not sure this fluent thing will stay forever, maybe don't mention it? I am even wondering why we keep QueryBuilders. Isn't it the same as Requests, which we said it's not that useful?

**Review by javanna — APPROVED** [review]
left two minors, LGTM otherwise

File: docs/java-rest/high-level/apis/queries.asciidoc (hunk 1 above)
Comment (tlrx) [inline docs/java-rest/high-level/apis/queries.asciidoc]: Sure.

File: docs/java-rest/high-level/apis/search.asciidoc (hunk 2 above)
Comment (tlrx) [inline docs/java-rest/high-level/apis/search.asciidoc]: > I am not sure this fluent thing will stay forever, maybe don't mention it?

Ok

> I am even wondering why we keep QueryBuilders

I think we should document it too. I think that people use `QueryBuilders` more than `Requests`. It's not that useful but it's everywhere, at least in our tests classes.

Comment (javanna) [inline docs/java-rest/high-level/apis/search.asciidoc]: yea it is probably our own fault if many people are using it :) no problem

**Comment by tlrx** [comment]
Thanks @javanna. I added all the search types and I really like the queries page. 

Sadly, the HTML link to QueryBuilders.multiMatchQuery() does not work because asciidoc replace `…` (&#8230) by `...` when processing the text and this changes the rendering of the HTML link. I tried to escape using \ but it does not work too. I think this can go in, and I'll ask Clinton if he knows how to deal with that.

**Review by javanna — APPROVED** [review]
this is great LGTM. Shall we do the same for aggs? :)

**Comment by tlrx** [comment]
Thanks @javanna 

> Shall we do the same for aggs? :)

I just forgot about them but good point... let's do that too.

**Pull request merged**